### DISCUSSION TITLE
What's the rule for the global DOFs in [https://ferrite-fem.github.io/Ferrite.jl/stable/examples/incompressible_elasticity/](url)

### DISCUSSION THREAD

**Discussion by zyhsjtu** [Ferrite-FEM/Ferrite.jl#983] [General]
in this example [https://ferrite-fem.github.io/Ferrite.jl/stable/examples/incompressible_elasticity/](url),
if my mesh is generated as 
`grid = generate_grid(Triangle, (nx, ny), Vec((0.0, 0.0)), Vec((2.0, 1.0)))`
the elements is (1, 2, 4), (2, 5, 4), ( 2, 3, 5) and  ( 3, 6, 5)
then, the global DOFs should be {1, 2, 3, 4, 5, 6,10,11,12}, {4, 5, 6,13,14,15, 10,11,12}, { 4, 5, 6,7,8,9,13,14,15} and {7,8,9,16,17,18,13,14,15}
but in the example [https://ferrite-fem.github.io/Ferrite.jl/stable/examples/incompressible_elasticity/](url), the global DOFs are [1, 2, 3, 4, 5, 6, 7, 8, 9], [3, 4, 10, 11, 5, 6, 8, 12, 9], [3, 4, 13, 14, 10, 11, 8, 15, 12] and [13, 14, 16, 17, 10, 11, 15, 18, 12].
What's the rule for the global DOFs?
and How can I get the global DOFs as this order {1, 2, 3, 4, 5, 6,10,11,12}, {4, 5, 6,13,14,15, 10,11,12}, { 4, 5, 6,7,8,9,13,14,15} and {7,8,9,16,17,18,13,14,15}?

**Comment by KnutAM** [comment]
In Ferrite, the DoFs are not ordered according to the node numbers. We are working on making this clearer in the docs, see https://github.com/Ferrite-FEM/Ferrite.jl/pull/875

The distribution algorithm is described in the devdocs [here](https://ferrite-fem.github.io/Ferrite.jl/dev/devdocs/dofhandler/#Ferrite.__close!) (note, this refers to the current master branch of Ferrite, so some vocabulary is slightly different). There is a method to `reorder!` the dofs as we discussed [here](https://github.com/Ferrite-FEM/Ferrite.jl/issues/892), but I think getting the permutation you ask for might be somewhat challenging to implement, and I'm not sure if it will break some used to enforce periodic constraints.

**Reply by zyhsjtu** [reply]
Thanks for your response, I had understood the rule and know how to deal with it.
